db/external_sst_file_basic_test.cc

```diff
@@ -162,7 +162,9 @@ class ExternalSSTFileBasicTest
         write_global_seqno, verify_checksums_before_ingest, true_data);
   }
 
-  ~ExternalSSTFileBasicTest() override { DestroyDir(env_, sst_files_dir_); }
+  ~ExternalSSTFileBasicTest() override {
+    EXPECT_OK(DestroyDir(env_, sst_files_dir_));
```

Comment:
EXPECT_OK doesn't throw, which probably would have made it a simpler choice for #7754

Reply:
It all depends on whether `GTEST_THROW_ON_FAILURE` is set or not. If it is (like in RocksDB's case), both `EXPECT_OK` and `ASSERT_OK` throw; otherwise neither throws.

Reply:
Oh super weird. It's an environment variable set in our Makefile, so NOT set when I'm running individual RocksDB unit tests. The setting appears original to gtest integration with no indication of why it is used. Do you know why?

Reply:
Nope. I did the same research as well the other day but like you said, it's been there since day 1. Maybe something in our continuous test infra relies on those exceptions getting thrown but I'm not sure.

Reply:
Maybe we should try unset `GTEST_THROW_ON_FAILURE`, default is false.

We unset it in circleci, so it could generate output file even the test fails:
https://github.com/facebook/rocksdb/blob/6cacb0d3a146604a432a06847234b78db096e198/.circleci/config.yml#L37